src/logging.c

```diff
@@ -63,6 +64,14 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
 {
   rcutils_ret_t ret = RCUTILS_RET_OK;
   if (!g_rcutils_logging_initialized) {
+    if (!rcutils_allocator_is_valid(&allocator)) {
+      allocator = rcutils_get_default_allocator();
+      RCUTILS_SET_ERROR_MSG(
+        "Provided allocator is invalid. Using the default allocator.", allocator);
```

Comment:
I actually feel like this should just case the function to fail, not fallback to the default allocator. If the allocator object itself is invalid, then that is a logic error in the program.

Reply:
Yeah, the "fallback" behaviour in general was put there because I was worried that the logging macros repeatedly calling `RCUTILS_LOGGING_AUTOINIT` and ignoring the return code would cause hard failures to go unnoticed. However, that argument doesn't apply here since the autoinit macro won't pass a custom allocator anyway, so I will remove this fallback to the default allocator and it is the caller's responsibility to check the return code.

Reply:
ac66eced2f3c381a887987ddb65af2d0b71017a4 removes the fallback